### PR TITLE
include an 'installMethod' property in server telemetry

### PR DESCRIPTION
**Pull request by ansuz**
This PR depends on [this commit](https://github.com/xwiki-labs/cryptpad/commit/f5e91ef3ef4c34c97178825c06fec2e81808d741) in the platform repository which will be merged into the 4.8.0 release.

It rewrites the `installMethod` attribute which has been added to the example config such that server telemetry indicates the install method that was used to create the instance (`"docker"` or `"docker-alpine"`).

This is in response to #25 

It also corrects a typo, changing the use of _"Wether"_ (Noun: a castrated ram) to _"Whether"_ (Conjunction: expressing a doubt or choice between alternatives).

### CODE REVIEW

**Comment by Asharas** [comment]
> It rewrites the installMethod attribute which has been added to the example config such that server telemetry indicates the install method that was used to create the instance ("docker" or "docker-alpine").  

Do the nginx based images need to be changed too?  
If yes, would `"docker-nginx"` and `"docker-nginx-alpine`" be appropriate labels?  
Do you want me to make that change (since I wrote those two images) or do you want to do it (because it's your PR)?

> It also corrects a typo, changing the use of _"Wether"_ (Noun: a castrated ram) to _"Whether"_ (Conjunction: expressing a doubt or choice between alternatives).

Oopsie :sweat_smile:

**Comment by ansuz** [comment]
> Do the nginx based images need to be changed too?

Hi @Asharas! this shouldn't be necessary as the relevant telemetry is sent by the application server (unless disabled).

> Oopsie

This is one of my favourite typos :smiley:

**Comment by Asharas** [comment]
> Hi @Asharas! this shouldn't be necessary as the relevant telemetry is sent by the application server (unless disabled).

The nginx based images are built from different Dockerfiles (`Dockerfile-nginx` and `Dockerfile-nginx-alpine`) so this PR changes won't be applied to nginx based images, hence my question on adding a similar `sed` line to them.

Also, I checked the `config.example.js` file on both this repo and on CryptPad's repo main branches and didn't find any line with `installMethod` (Although it is not really a problem since `sed` shouldn't throw an error if the pattern is absent).  
I assume it is an upcoming change?

**Comment by ansuz** [comment]
> The nginx based images are built from different Dockerfiles (Dockerfile-nginx and Dockerfile-nginx-alpine) so this PR changes won't be applied to nginx based images, hence my question on adding a similar sed line to them.

I see! I'm not very familiar with docker and I misinterpreted the config files. I assumed that multiple docker images were launched with the relevant nginx files proxying to a corresponding application server in a different image (I think that's how docker-compose works?).

In that case, yes, it would be appropriate to add a modified string for these configurations as well. Based on the telemetry we do receive we can see that a large number of instances use configurations that don't apply some important security settings. Ideally this new parameter would help identify if one method or another is more commonly misconfigured so we can figure out why and correct the root cause.

> Also, I checked the `config.example.js` file on both this repo and on CryptPad's repo main branches and didn't find any line with `installMethod` (Although it is not really a problem since `sed` shouldn't throw an error if the pattern is absent).
> I assume it is an upcoming change?

That's correct. The new parameter is in `config.example.js` on the `staging` branch, which we'll merge to `main` for the `4.8.0` release (or `4.7.1` if we decide to publish an intermediary release).

**Comment by ansuz** [comment]
Unfortunately I don't see any other references to the config file in either `Dockerfile-nginx` or `Dockerfile-nginx-alpine`, which is why I assumed they were unrelated initially. As such I'm not sure what path to use for the `sed` call.

**Comment by Asharas** [comment]
> Unfortunately I don't see any other references to the config file in either `Dockerfile-nginx` or `Dockerfile-nginx-alpine`, which is why I assumed they were unrelated initially. As such I'm not sure what path to use for the `sed` call.

TL;DR You're right, I've been mistaking.  

To give more detail, I primarily wrote those image because, in a Docker Swarm cluster, it is not possible to share a volume between containers as they could spawn on different node, in that case having the webserver serve html content and the rest by the node server.  
For example this is possible when running Docker Compose on a standalone node, which is different than Docker Swarm, although they use almost the same configuration syntax with only some options not available in one mode or the other (eg: you can't use the `build` directive in Swarm mode, images have to be pre-built or pulled from a library and you can't use `docker config` or `docker secret` if not in a Swarm.)  
To remedy that and still have two different containers would require to build two images, one nginx and one node, but with the same content, hence duplicated.

To automate the deployment these images configure the web server through environment variables via the `docker-entrypoint.sh` script.  
I originally intended to set some of the options in the `config.js` in the same way but as of now this part is commented and not in use (because of the admin and support mailbox keys that require manual config after instance first launch and to avoid instances admins not reading the `config.js`).  
So by using those images `config.js` must be provided externally, eg by mouting it into the container or, in my case, using `docker config` (a feature of Docker Swarm cluster to provide configs to containers in the cluster through docker compose directive).  

The downside of all that is that it removes the possibility of scaling up the web server independently from the node server.

**Comment by Asharas** [comment]
> because of the admin and support mailbox keys that require manual config after instance first launch and to avoid instances admins not reading the config.js  

Something else I'm thinking about: the current standalone node containers `Dockerfile` and `Dockerfile-alpine` only set the listening adress to `0.0.0.0` but not the `safeOrigin` and `unsafeOrigin`.  
@promasu , you once told me that you use the node version of this image with Kubernetes, am I right? How do you manage setting the origin options? Do you provide an external `config.js` through Kubernetes manifest?

If in this case as the using nginx based images `config.js` is provided externally then those `sed` lines don't seem necessary (aswell as those changing listening address).

**Comment by ansuz** [comment]
> Do the nginx based images need to be changed too?  
> If yes, would "docker-nginx" and "docker-nginx-alpine" be appropriate labels?  
> Do you want me to make that change (since I wrote those two images) or do you want to do it (because it's your PR)?  

Oops, I came back to see about this PR and realized I didn't answer your questions, @Asharas  :sweat: 

The labels you proposed seem good to me. I'm not sure what would be an appropriate path to the config in those files since there isn't an existing sed call that I can mimic.

It would be nice have the same style labels applied everywhere, but I don't think it's worth blocking this PR to have them. Maybe it's best to just merge as-is and make that another PR?

> If in this case as the using nginx based images `config.js` is provided externally then those `sed` lines don't seem necessary (aswell as those changing listening address).

In any case, if the file doesn't exist or the pattern doesn't occur within it (as in older config files) it won't do any harm, so this doesn't seem like a blocking issue either.

**Comment by ansuz** [comment]
Since there've been no objections I'm gonna go ahead and merge for tomorrow's release.
